plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -158,14 +169,33 @@
     private static final long MIN_SUPPORTED_DATE_EPOCH = LocalDate.parse("1970-01-01").toEpochDay();
     private static final long MAX_SUPPORTED_DATE_EPOCH = LocalDate.parse("2106-02-07").toEpochDay(); // The max date is '2148-12-31' in new ClickHouse version
 
-    private static final LocalDateTime MIN_SUPPORTED_TIMESTAMP = LocalDateTime.parse("1970-01-01T00:00:00");
-    private static final LocalDateTime MAX_SUPPORTED_TIMESTAMP = LocalDateTime.parse("2105-12-31T23:59:59");
-    private static final long MIN_SUPPORTED_TIMESTAMP_EPOCH = MIN_SUPPORTED_TIMESTAMP.toEpochSecond(UTC);
-    private static final long MAX_SUPPORTED_TIMESTAMP_EPOCH = MAX_SUPPORTED_TIMESTAMP.toEpochSecond(UTC);
+    // For ClickHouse DateTime/DateTime64 types, different ClickHouse versions support different min/max values
+    // before 21.4
+    //   DateTime   min value: 1970-01-01 00:00:00, max value: 2106-02-06 06:28:15
+    //   DateTime64 min value: 1970-01-01 00:00:00, max value: 2106-02-06 06:28:15.999999999
+    // after 21.4
```

Comment:
We can't realize the range in exact version 21.4. Maybe `after or equal to 21.4` is better.